dips/dip-158.md

```diff
@@ -0,0 +1,775 @@
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|given_name|str|Y|Legal given name of the payer|
+|surname|str|Y|Legal surname of the payer|
+|address|[AddressObject](#addressobject)|Y|Merchant's physical address information|
+|national_id|[NationalIdObject](#nationalidobject)|N|National ID information for payer. e.g. SSN|
+
+```
+"payer_data": 
+{
+  "name": "Alice",
+  "surname": "Cooper",
+  "address": {
+    AddressObject()
+  },
+  "national_id": {
+    NationalIdObject()
+  }        
+}
+``` 
+## BusinessDataObject
+The details of the merchant.
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|name|str|Y|Merchant's display name. Should be recognizable by the payer|
+|legal_name|str|Y|The legal entity name|
+|image_url|str|N|URL with the business logo|
+|address|[AddressObject](#addressobject)|Y|Merchant's physical address information|
+```
+"business_data": 
+{
+  "name": "Acme",
+  "legal_name": "Acme Technologies Inc.",
+  "address": {
+    AddressObject()
+  }        
+}
+``` 
+
+## PaymentActionObject
+Information about the payment type (i.e. action), amount and currency.
+
+|Field|Type|Required?|Description|
+|-----|----|---------|-----------|
+| amount | uint | Y | Amount of the transfer.  Base units are the same as for on-chain transactions for this currency.  For example, if DiemUSD is represented on-chain where "1" equals 1e-6 dollars, then "1" equals the same amount here.  For any currency, the on-chain mapping must be used for amounts. |
+| currency | enum | Y | One of the supported on-chain currency types, e.g. XUS|
+| action | [Action enum](#action-enum) | Y | Populated in the request.  This value indicates the requested action to perform. For immediate capture and P2P transfer, "charge" is used.  For auth and capture, "auth" and "capture" should be used. "capture" can only be performed after a valid "auth" |
+| valid_until | uint | N | Unix timestamp indicating until when the payment can be authorized. Once this time has been reached, an attempt to authorize and lock the funds will be rejected. One should consider limiting this period to several minutes (e.g. 30 minutes) which should suffice for the customer to approve or reject the payment request|
+| timestamp | uint | Y | Unix timestamp indicating the time that the payment Command was created.|
+
+```
+{
+    "amount": 100000000,
+    "currency": "XUS",
+    "action": "auth",
+    "valid_until": 74000,
+    "timestamp": 72322,
+}
+```
+### Action Enum
+The following values are allowed for the Action Enum
+* `charge`
+* `auth`
+* `capture`
+
+## AddressObject
+Represents a physical address
+| Field | Type | Required? | Description |
+|-|-|-|-|
+| city | str | Y | The city, district, suburb, town, or village |
+| country | str | Y | Two-letter country code (https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) |
+| line1 | str | Y | Address line 1 |
```

Comment:
Let's discuss whether or not we require line1.

Reply:
Decided it will be mandatory